appserver/tests/admin/tests/src/test/java/org/glassfish/main/admin/test/progress/ProgressStatusComplexITest.java

```diff
@@ -67,9 +67,9 @@ public void executeCommandWithSupplements() {
         assertArrayEquals(new String[] {
             "Starting", "2 seconds supplemental command", "Parsing",
             "Working on main part", "Finished", "3 seconds supplemental" },
-            uniqueMessages(prgs));
+                uniqueMessages(prgs));
 
-        assertThat(prgs, hasSize(21));
+        assertThat(prgs, hasSize(greaterThanOrEqualTo(21)));
```

Comment:
Maybe use the uniqueMessages for the check exact number, not sure, I am just reviewing using github. I am not sure what exactly uniqueMessages does.

Reply:
Exact number isn’t very useful. The log contains lines like:

20% done
35% done
80% done
….

When the system is slower, it will just produce more lines, like:

15% done
28% done
40% done
65% done
…

So the behaviour is not deterministic. It would be better to check the result in a different way but I didn’t want to bother.

Reply:
I'm not sure how it is computed, I have to take a look. I would expect it is deterministic, but it can skip some lines. But then it is suspicious that it changes behavior with this PR and it didn't fail before.

Reply:
It’s not suspicious. It just doesn’t pass on my computer, when I run it together with other tests. It passes on my computer when I run the test alone. I can revert it and see if it still passes on Jenkins.

But the assertion is clearly non-deterministic.

Reply:
Integration tests should not run in parallel with other tests. The problem with these tests is that they strictly follow some scenario, once you allow them to be more tolerant, they tolerate real bugs. I hope you didn't find some :-)
Btw these "asadmin commands with progress and extensions" are unused feature, but maybe they could be interesting in the future, ie. if GF would want to support something like "ear/war installers" in cloud ... but I am not sure if it would be even useful today ... just intuition.

Reply:
I didn't run anything in parallel. But it seems that GlassFish or my computer slows down when I execute all the test sequentially. When I executed only the failed test, it passed.

Anyway, I reverted my changes in ProgressStatusComplexITest now. Myabe it was failing only on my computer and it will pass on Jenkins. In the future, we should make the assertions more robust, I'm pretty sure that the test isn't deterministic and might fail on slower systems.

Reply:
Hmm, there might be some issue, it should not slow down or cause these issues. I will take a look, we have yet one suspicious test, where I still did not find the cause, but it probably depends on order of tests.

We should make GlassFish robust, not make tests tolerant to issues, we improved it a lot, but we did no finish yet. Those percents are not random numbers. However it is a test with threads, so it is not simple to debug it.